src/modules/metrics/metrics.controller.ts

```diff
@@ -0,0 +1,39 @@
+import { ApiTags } from '@nestjs/swagger';
+import { Controller, Get, Inject } from '@nestjs/common';
+import { Registry } from 'prom-client';
+import { Public } from '../authentication/api/public.decorator.js';
+import { ReporterService } from './reporter/reporter.service.js';
+import { RollenArt } from '../rolle/domain/rolle.enums.js';
+import { DBiamPersonenkontextRepo } from '../personenkontext/persistence/dbiam-personenkontext.repo.js';
+
+@ApiTags('Metrics')
+@Controller({ path: 'metrics' })
+export class MetricsController {
+    public constructor(
+        @Inject(Registry)
+        private readonly registry: Registry,
+        @Inject(ReporterService)
+        private readonly reporterService: ReporterService,
+        private readonly dBiamPersonenkontextRepo: DBiamPersonenkontextRepo,
+    ) {}
+
+    @Get()
+    @Public()
+    public async getMetrics(): Promise<string> {
+        const mapping: Map<string, RollenArt> = new Map([
+            ['number_of_teachers', RollenArt.LEHR],
+            ['number_of_students', RollenArt.LERN],
+            ['number_of_admins', RollenArt.LEIT],
+        ]);
+
+        await Promise.all(
+            Array.from(mapping).map(async ([metric, rolle]: [string, RollenArt]) => {
+                const count: number = await this.dBiamPersonenkontextRepo.getPersonCountByRole(rolle);
+                this.reporterService.gauge(metric, count, { school: 'all' });
```

Comment:
.gauge() isn't even a promise so it's not asynchronous, no point awaiting it inside a promise.all.

I'd do it this way:

Object instead of map for less complexity:
`    const mapping: { [metric: string]: RollenArt } = {
        number_of_teachers: RollenArt.LEHR,
        number_of_students: RollenArt.LERN,
        number_of_admins: RollenArt.LEIT,
    };`

then I'd just loop through the objects entries, await the count and then gauge the count:

  `for (const [metric, rolle] of Object.entries(mapping)) {
        const count: number = await this.dBiamPersonenkontextRepo.getPersonCountByRole(rolle);
        this.reporterService.gauge(metric, count, { school: 'all' });
    }`

Reply:
Yes thats why we await for it to be resolved. .gauge() isn't async though.

Reply:
Nvm!